src/inc/msquichelper.h

```diff
@@ -1020,4 +1019,24 @@ WriteSslKeyLogFile(
 #endif
 }
 
+struct StrBuffer
+{
+    uint8_t* Data;
+    uint16_t Length;
+
+    StrBuffer(const char* HexBytes)
+    {
+        Length = (uint16_t)(strlen(HexBytes) / 2);
+        Data = new uint8_t[Length];
+
+        for (uint16_t i = 0; i < Length; ++i) {
+            Data[i] =
+                (DecodeHexChar(HexBytes[i * 2]) << 4) |
+                DecodeHexChar(HexBytes[i * 2 + 1]);
+        }
+    }
+
+    ~StrBuffer() { delete [] Data; }
```

Comment:
Rule of 3/5? Are resources/RAII not used in MsQuic?

Reply:
Ah, this is just relocated code.

Reply:
No, not really. And I prefer not to go down that rabbit hole right now. I already improved things by deduplicating this...